raiden/network/proxies/token_network.py

```diff
@@ -747,13 +761,18 @@ def set_total_deposit(
         checking_block = self.client.get_checking_block()
         error_prefix = 'setTotalDeposit call will fail'
         with self.channel_operations_lock[partner], self.deposit_lock:
-            amount_to_deposit, log_details = self._deposit_preconditions(
-                channel_identifier=channel_identifier,
-                total_deposit=total_deposit,
-                partner=partner,
-                token=token,
-                block_identifier=given_block_identifier,
-            )
+            try:
+                amount_to_deposit, log_details = self._deposit_preconditions(
```

Comment:
`amount_to_deposit` and `log_details` are initially declared/initialized here. If the precondition raises, then the values will not be initialized and therefore, line 790 and 819 will fail with a "used before intialized" kind of error.

I guess it's pretty much the same case for all the places you added a try..catch..pass.

Also, could you please add a test case for a failure in the precondition for each of those methods? The tests passed though the above is an issue that wasn't caught by our tests so far.